actors/market/src/lib.rs

```diff
@@ -190,14 +190,17 @@ impl Actor {
         struct ValidDeal {
             proposal: DealProposal,
             cid: Cid,
-            allocation: AllocationID,
         }
-
         // Deals that passed validation.
         let mut valid_deals: Vec<ValidDeal> = Vec::with_capacity(params.deals.len());
         // CIDs of valid proposals.
         let mut proposal_cid_lookup = BTreeSet::new();
         let mut total_client_lockup: BTreeMap<ActorID, TokenAmount> = BTreeMap::new();
+        // Client datacap balance remaining after allocations for deals processed so far.
+        let mut client_datacap_remaining: BTreeMap<ActorID, TokenAmount> = BTreeMap::new();
```

Comment:
I am mildly interested in modifying provider and client lockup accounting to the same model of tracking remaining instead of total.  Conceptually all three can be unified into `track balance -> check enough -> deduct from remaining` which will make understanding this code a little easier. We could maybe even put all three checks behind the same interface and also share some of the implementation.

I'm not going to block the merge on this but leaving this for your thoughts.  If you agree I'll make a low priority cleanup issue.

Reply:
An issue sounds good, thanks